### PR TITLE
Update the long pole

### PR DESCRIPTION
**Pull request by bloodbowel**
#### Summary
Features "Updates the long pole"

#### Purpose of change
I noticed the recipe for the long pole is similar to the spear shaft in some ways. You could cut the period where the player watches oil soak into the wood. I also updated it's to-hit calculation and made it's construction more realistic using a long wooden post rather than a complete wooden beam.

#### Describe the solution
1. In #75640 I removed 4 hours from construction time by adding a soaking period. I did the same thing for the long pole here because the recipe is similar. Now those 4 hours are spent soaking while the player can go do something else.
2. I noticed that the long pole uses the old "to-hit: -1" to derive it's to-hit value. I updated this to use the modern grip/length/surface/balance calculation.
3. I noticed this item is created from the wooden beam which is way to thick for it. It now uses a long wooden post which is much thinner but not too thin. I increased it's recipe time to 2 hours and 40 minutes to reflect all the time needed to cut it into shape. That means to create this now takes 2 hours and 40 minutes in construction and 4 hours in a soaking state. So 6 hours and 40 minutes total.

#### Describe alternatives you've considered
Leave the long pole alone. Maybe adopt some of these changes and leave others.

#### Testing
Tested on my computer in debug mode.

#### Additional context
Screenshot

### CODE REVIEW

File: data/json/items/resources/wood.json [hunk 1]
```diff
@@ -72,13 +72,42 @@
     "category": "spare_parts",
     "melee_damage": { "bash": 18 }
   },
+  {
+    "type": "GENERIC",
+    "id": "soaking_long_pole",
+    "name": { "str_sp": "soaking long pole" },
+    "description": "A stout, ten-foot pole.  Could be used similarly to a spear. It's currently being soaked with oil and quite slippery. Once the oil penetrates the wood it can be wiped down and used for further projects.",
```
Comment (TheShadowFerret) [inline data/json/items/resources/wood.json:79]:
```suggestion
    "description": "A stout, ten-foot pole.  Could be used similarly to a spear. It's currently soaked with oil and quite slippery. Once the oil penetrates the wood it can be wiped down and used for further projects.",
```

**Comment by Maleclypse** [comment]
./tests/item_test.cpp:757: FAILED:
(~[slow] ~[.],starting_items)=>   REQUIRE( msg_prune.empty() )
(~[slow] ~[.],starting_items)=> with expansion:
Error: (~[slow] ~[.],starting_items)=>   false
(~[slow] ~[.],starting_items)=> with messages:
(~[slow] ~[.],starting_items)=>   msg_enforce := ""
(~[slow] ~[.],starting_items)=>   msg_prune := "
(~[slow] ~[.],starting_items)=>   [
(~[slow] ~[.],starting_items)=>     "long_pole"
(~[slow] ~[.],starting_items)=>   ]
(~[slow] ~[.],starting_items)=>   The item(s) above should be removed from the blacklist at /data/mods/
(~[slow] ~[.],starting_items)=>   TEST_DATA/legacy_to_hit.json."


And soaking long pole is a weapon but wasn't added to the expected dps test

**Comment by Maleclypse** [comment]
../tests/effective_dps_test.cpp:294: FAILED:
(~[slow] ~[.],starting_items)=>   CHECK( calc_expected_dps( weap.first ) == weap.second )
(~[slow] ~[.],starting_items)=> with expansion:
Error: (~[slow] ~[.],starting_items)=>   Approx( 6.514137581 ) == 7.4
(~[slow] ~[.],starting_items)=> with message:
(~[slow] ~[.],starting_items)=>   soaking_long_pole's dps changed, if it's intended replace the value in the
(~[slow] ~[.],starting_items)=>   respective file in data/mods/TEST_DATA/expected_dps_data.

**Comment by bloodbowel** [comment]
What are the stats of the character for the test? I'm supposed to estimate an average dps but I'm not sure how.

**Comment by Night-Pryanik** [comment]
> What are the stats of the character for the test?

It's 10/10/10/10 guy with 4 in all melee skills.
https://github.com/CleverRaven/Cataclysm-DDA/blob/92ef2d67770ee628e60d53c9f6d9a215eda5f187/tests/effective_dps_test.cpp#L240-L258

File: data/json/items/resources/wood.json [hunk 2]
```diff
@@ -72,13 +72,42 @@
...
+    },
+    "price_postapoc": "50 cent",
+    "qualities": [ [ "HAMMER", 1 ] ],
+    "melee_damage": { "bash": 14 }
+  },
   {
     "type": "GENERIC",
     "id": "long_pole",
-    "name": { "str": "long pole" },
+    "name": { "str_sp": "long pole" },
```
Comment (Night-Pryanik) [inline data/json/items/resources/wood.json:108]: Why this change? Plural of pole is poles, I think?

Comment (bloodbowel) [reply data/json/items/resources/wood.json:108]: I don't remember why that was changed. It should be just "str", since you can put an s at the end for the plural I think.

Comment (Night-Pryanik) [reply data/json/items/resources/wood.json:108]: Yes, it should be str.

Comment (Night-Pryanik) [reply data/json/items/resources/wood.json:108]: Yes, it should be str.

**Comment by bloodbowel** [comment]
So that's what that value meant. I feel like an idiot. Oh well, now it passed. I was testing it in debug mode and looking at the item description for the dps, but it was incorrect even with the character set to be like the debug man.

But now I am getting this error:
Mods-(~[slow] ~[.],starting_items)=> ../tests/vehicle_fake_part_test.cpp:363: FAILED:
Mods-(~[slow] ~[.],starting_items)=>   CHECK( !you.sees( you.pos_bub() + point( 10, 10 ) ) )
Mods-(~[slow] ~[.],starting_items)=> with expansion:
Error: Mods-(~[slow] ~[.],starting_items)=>   false

What is that? I didn't mess with any vehicle parts.

**Comment by Night-Pryanik** [comment]
This error isn't related to your PR.
